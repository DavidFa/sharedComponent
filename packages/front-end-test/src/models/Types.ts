export interface PostType {
    id: number;
    title: string;
    body: string
}

export enum PostStatus {
    list,
    edit,
    add
}