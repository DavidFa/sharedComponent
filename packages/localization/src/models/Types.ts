export type InputFormModel = {
    name: InputFormProps;
    description: InputFormProps;
}


export type InputFormProps = {
    label: string;
    placeholder: string;
    length: number;
}

export type InputFormWithLanguageProps = {
    language: string;
    placeholder: string;
    length: number;
}

export interface LocalizationState {
    language: string;
    languages: string[];
}

export enum LocalizationActionType {
    updateLanguage,
    populateLanguages
}

export type LocalizationAction = {
    type: LocalizationActionType;
    payload: {
        language?: string;
        languages?: string[];
    }
}