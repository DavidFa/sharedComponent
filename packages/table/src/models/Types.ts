export type DataGridColumns = {
    field: string;
    headerName: string;
    width: number;
    align?: string;
    sortable?: boolean;
}

export type DataGridRows = {
    [key: string]: string | number
}