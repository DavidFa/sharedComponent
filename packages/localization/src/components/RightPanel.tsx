import React from "react";
import styled from "styled-components";
import useRightPanel from "./hooks/useRightPanel";

const Panel = styled.div`
margin: 10px;
padding: 20px;
background: #fff;
box-shadow: 0 6px 20px 10px #ccc;
border-radius: 6px;
`;

const Form = styled.form`
`;

const Button = styled.button.attrs({ type: "button" })`
padding: 5px 10px;
margin: 0 10px;
`;

const H4 = styled.h4``;

const Wrapper = styled.div`
text-align:right;
margin: 20px 0 0 0;
`;


const RightPanel: React.FC = () => {


    const { flyOutField, inputs, dissFlyoutHandler, onSubmit } = useRightPanel();

    // const inputs = useMemo(() => {
    //     languages?.map((item: string, index: number) => {
    //         const value = item === language ? field.tem : fieldSet[item] || '';
    //         return <InputFormWithLanguage key={`${item}_${index}`} language={item} placeholder={item} length={field.length} value={value} changeField={onChangeFieldHandler} />
    //     });
    // }, [flyOutField, name, description]);


    return <Panel>
        <Form>
            <H4 role="h4">{flyOutField}</H4>
            {inputs}
            <Wrapper>
                <Button data-testid="cancelBtn" onClick={dissFlyoutHandler}>Cancel</Button>
                <Button data-testid="saveBtn" onClick={onSubmit}>Save</Button>
            </Wrapper>
        </Form>
    </Panel>
}

export default RightPanel;