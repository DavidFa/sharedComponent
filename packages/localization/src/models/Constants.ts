export const FIELD_NAME: string = "Name";
export const FIELD_DESC: string = "Description";