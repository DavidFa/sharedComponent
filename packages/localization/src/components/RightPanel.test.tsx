import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import RightPanel from "./RightPanel";
import { ActionType } from "../models/Types";
// import { cleanup, fireEvent, render } from '@testing-library/react';
import { Store, AnyAction } from 'redux';
// import {screen, getByLabelText} from '@testing-library/dom'
import { cleanup, fireEvent, render, screen } from '@testing-library/react'

const mockStore = configureStore([]);

describe('RightPanel', () => {

    // Snapshot
    it('should render with given state from Redux store', () => {
        const store: Store<any, AnyAction> = mockStore({
            language: {
                language: "en-AU",
                languages: ["en-AU", "en-US", "en", "en-UK"]
            },
            localization: {
                isShowFlyout: false,
                flyOutField: "Name",
                name: {
                    length: 20,
                    tem: "",
                    localization: {}
                },
                description: {
                    length: 50,
                    tem: "",
                    localization: {}
                }
            },
        });
        const component: renderer.ReactTestRenderer = renderer.create(
            <Provider store={store}>
                <RightPanel />
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });


    // afterEach(cleanup);
    // it('should display header', () => {
    //     const h4 = screen.getByRole("form");
    // })


    const store: Store<any, AnyAction> = mockStore({
        language: {
            language: "en-AU",
            languages: ["en-AU", "en-US", "en", "en-UK"]
        },
        localization: {
            isShowFlyout: false,
            flyOutField: "Name",
            name: {
                length: 20,
                tem: "",
                localization: {}
            },
            description: {
                length: 50,
                tem: "",
                localization: {}
            }
        },
    });

    beforeEach(() => {

        store.dispatch = jest.fn();
        render(<Provider store={store}>
            <RightPanel />
        </Provider>)
    });

    it('should dispatch an action on Cancel button click', () => {

        // Dom
        const cancelBtn = screen.getByTestId("cancelBtn");
        expect(cancelBtn).toHaveTextContent("Cancel");

        // Event
        fireEvent.click(cancelBtn);
        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(
            { type: ActionType.dissFlyout, payload: { local: "en-AU" } }
        );
    });

    // afterEach(cleanup);

    it('should contain h4 elements', () => {

        const h4 = screen.getByRole('h4');
        expect(h4).toHaveTextContent("Name");

    });

    it('should contain 4 input elements', () => {

        const inputs = screen.queryAllByRole('textbox');
        expect(inputs).toHaveLength(4);

        const inputAu = screen.getByPlaceholderText("en-AU");
        expect(inputAu).toBeInTheDocument();

        const inputUs = screen.getByPlaceholderText("en-US");
        expect(inputUs).toBeInTheDocument();

        const inputUk = screen.getByPlaceholderText("en-UK");
        expect(inputUk).toBeInTheDocument();

        const inputEn = screen.getByPlaceholderText("en");
        expect(inputEn).toBeInTheDocument();
    });

    
    it('should dispatch an action on Save button click', () => {

        // Dom
        const saveBtn = screen.getByTestId("saveBtn");
        expect(saveBtn).toHaveTextContent("Save");

        // Event
        fireEvent.click(saveBtn);
        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(
            { type: ActionType.updateNameSet, payload: { local: "en-AU", localization: { "en": "", "en-UK": "", "en-US": "" } } }
        );
    });
})
