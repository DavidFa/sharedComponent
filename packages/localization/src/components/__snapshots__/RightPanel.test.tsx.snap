// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightPanel should render with given state from Redux store 1`] = `
<div
  className="sc-hKgJUU hZjYMl"
>
  <form
    className="sc-eCstlR eIMPPg"
  >
    <h4
      className="sc-gKseQn"
      role="h4"
    >
      Name
    </h4>
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <input
        className="sc-dlfnuX idJtJP"
        maxLength={20}
        onChange={[Function]}
        placeholder="en-AU"
        value=""
      />
      <label
        className="sc-gsTEea cCGvIt"
      >
        en-AU
      </label>
    </div>
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <input
        className="sc-dlfnuX idJtJP"
        maxLength={20}
        onChange={[Function]}
        placeholder="en-US"
        value=""
      />
      <label
        className="sc-gsTEea cCGvIt"
      >
        en-US
      </label>
    </div>
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <input
        className="sc-dlfnuX idJtJP"
        maxLength={20}
        onChange={[Function]}
        placeholder="en"
        value=""
      />
      <label
        className="sc-gsTEea cCGvIt"
      >
        en
      </label>
    </div>
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <input
        className="sc-dlfnuX idJtJP"
        maxLength={20}
        onChange={[Function]}
        placeholder="en-UK"
        value=""
      />
      <label
        className="sc-gsTEea cCGvIt"
      >
        en-UK
      </label>
    </div>
    <div
      className="sc-iBPTik htTLnT"
    >
      <button
        className="sc-jSgvzq iPcuja"
        data-testid="cancelBtn"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="sc-jSgvzq iPcuja"
        data-testid="saveBtn"
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;
